src/mode/mode.ts

```diff
@@ -42,35 +49,97 @@ export abstract class Mode {
+        "$" : async (p, o) => { return p.getLineEnd(); },
+        "0" : async (p, o) => { return p.getLineBegin(); },
+        "^" : async (p, o) => { await vscode.commands.executeCommand("cursorHome"); return this._motion.position; },
+        "gg" : async (p, o) => {
+            return new Position(0, Position.getFirstNonBlankCharAtLine(0), p.positionOptions);
+        },
+        "<count>G" : async (p, o) => {
+             const lastLine = p.getDocumentEnd().line;
+             if (o.count === 0) {
+                 return new Position(lastLine, Position.getFirstNonBlankCharAtLine(lastLine), p.positionOptions);
+             } else {
+                 const newLine = Math.min(o.count - 1, lastLine);
+                 return new Position(newLine, Position.getFirstNonBlankCharAtLine(newLine), p.positionOptions);
+             }
+         },
+        "<count>w" : async (p, o) => {
+            return p.getWordRight(o.count); },
+        "<count>W" : async (p, o) => { return p.getBigWordRight(o.count); },
+        "<count>e" : async (p, o) => { return p.getCurrentWordEnd(o.count); },
+        "<count>E" : async (p, o) => { return p.getCurrentBigWordEnd(o.count); },
+        "<count>b" : async (p, o) => { return p.getWordLeft(o.count); },
+        "<count>B" : async (p, o) => { return p.getBigWordLeft(o.count); },
+        "<count>}" : async (p, o) => { return p.getCurrentParagraphEnd(o.count); },
+        "<count>{" : async (p, o) => { return p.getCurrentParagraphEnd(o.count); },
+        "<count><c-f>": async (p, o) => { await vscode.commands.executeCommand("cursorPageDown"); return this._motion.position; },
+        "<count><c-b>": async (p, o) => { await vscode.commands.executeCommand("cursorPageUp"); return this._motion.position; },
+        "%" : async (p, o) => {
+             const oldPosition = this._motion.position;
+             await vscode.commands.executeCommand("editor.action.jumpToBracket");
+             if (oldPosition === this._motion.position) {
+                 return this._motion.position;
+             } else {
+                 return this._motion.position.getLeft();
+             }
         },
-        "$" : async (c) => { return c.getLineEnd(); },
-        "0" : async (c) => { return c.getLineBegin(); },
-        "w" : async (c) => { return c.getWordRight(); },
-        "e" : async (c) => { return c.getCurrentWordEnd(); },
-        "b" : async (c) => { return c.getWordLeft(); },
-        "}" : async (c) => { return c.getCurrentParagraphEnd(); },
-        "{" : async (c) => { return c.getCurrentParagraphBeginning(); }
+
+        "<count>t<argument>" : async (p, o) => { return p.tilForwards(o.argument, o.count); },
+        "<count>T<argument>" : async (p, o) => { return p.tilBackwards(o.argument, o.count); },
+        "<count>f<argument>" : async (p, o) => { return p.findForwards(o.argument, o.count); },
+        "<count>F<argument>" : async (p, o) => { return p.findBackwards(o.argument, o.count); }

```

Comment:
Great : This pull request implements https://github.com/VSCodeVim/Vim/issues/187